src/IDCodeValidator.java

```diff
@@ -0,0 +1,219 @@
+            if (isGenderNumberCorrect(personalCode.genderNumber)) {
+                if (isDayNumberCorrect(personalCode.day)) {
+                    if (isMonthNumberCorrect(personalCode.month)) {
+                        if (isBirthDateCorrect(personalCode.year, personalCode.month, personalCode.day)) {
+                            if (checkControlNumber(idCode)) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+        } else {
+            System.out.println("The length doesn't match the rules");
+            return false;
+        }
+        System.out.println("The Id code doesn't match the rules");
+        return false;
+
+    }
+
+    /**
+     * Проверяет корректность кода пола в ID code.
+     *
+     * @param genderNumber целое число, представляющее код пола
+     * @return {@code true}, если код пола корректен, иначе {@code false}
+     */
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (Gender.isGenderNumberCorrect(genderNumber)) {
+            System.out.println("Gender number is correct");
+            return true;
+        }
+        System.out.println("Gender number is not correct");
+        return false;
+    }
+
+
+    /**
+     * Проверяет корректность номера месяца рождения в ID code.
+     *
+     * @param dayNumber целое число, представляющее номер месяца
+     * @return {@code true}, если номер дня корректен, иначе {@code false}
+     */
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber < 32) {
+            System.out.println("Day number is correct");
+            return true;
+        }
+        System.out.println("Day number is not correct");
+        return false;
+    }
+
+    /**
+     * Проверяет корректность номера месяца рождения в ID code.
+     *
+     * @param monthNumber целое число, представляющее номер месяца
+     * @return {@code true}, если номер месяца корректен, иначе {@code false}
+     */
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+
+        if (monthNumber > 0 && monthNumber < 13) {
+            System.out.println("Month number is correct");
+            return true;
+        }
+        System.out.println("Month number is not correct");
+        return false;
+    }
+
+
+    /**
+     * Проверяет корректность даты рождения в ID code.
+     *
+     * @param yearNumber  целое число, представляющее год рождения (1955, 1999, 2013)
+     * @param monthNumber целое число, представляющее номер месяца (1, 8)
+     * @param dayNumber   целое число, представляющее номер дня (15, 31)
+     * @return {@code true}, если номер дня корректен, иначе {@code false}
+     */
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        boolean result ;
+        HashMap<Integer, Integer> monthToDays = new HashMap<>();
```

Comment:
We haven't learned that yet but the convention is to use Interface as a type and not the implementation
So, `Map<Integer, Integer> monthToDays = new HashMap<>();`